jib-core/src/main/java/com/google/cloud/tools/jib/configuration/PortsWithProtocol.java

```diff
@@ -79,6 +79,30 @@ public Protocol getProtocol() {
     return protocol;
   }
 
+  @Override
+  public boolean equals(Object other) {
+    if (other == this) {
+      return true;
+    }
+    if (other == null || other.getClass() != PortsWithProtocol.class) {
```

Comment:
Curious. Is there a reason to prefer this over `!other instanceof PortsWithProtocol`? Because of inheritance?

Reply:
Honestly I used it because I forgot `instanceof` exists. So I'm not sure.

Reply:
We should probably use `instanceof` here.